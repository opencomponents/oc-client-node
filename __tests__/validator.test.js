'use strict';

const validator = require('../src/validator');

test('validator api', () => {
  expect(validator).toHaveProperty('validateConfiguration'); // Q.: why an object arount this only function then?
  expect(validator).toMatchSnapshot();
});

const scenarios = [
  {
    description: 'conf is an empty object',
    conf: {},
    isValid: true // Q.: why this is valid?
  },
  {
    description: 'conf.registries is an empty array',
    conf: { registries: [] },
    isValid: false
  },
  {
    description: 'conf.registries is an empty object',
    conf: { registries: {} },
    isValid: false
  },
  {
    description: 'conf.registries is a valid object',
    conf: { registries: { serverRendering: 'http://registry.local:3030' } },
    isValid: true
  }
];

scenarios.forEach(({ description, conf, isValid }) => {
  test(description, () => {
    const result = validator.validateConfiguration(conf);
    expect(result.isValid).toEqual(isValid);
    expect(result).toMatchSnapshot();
  });
});
