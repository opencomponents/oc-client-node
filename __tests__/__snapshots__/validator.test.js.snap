// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conf is an empty object 1`] = `
Object {
  "isValid": true,
}
`;

exports[`conf.registries is a valid object 1`] = `
Object {
  "isValid": true,
}
`;

exports[`conf.registries is an empty array 1`] = `
Object {
  "error": "Configuration is not valid: registries must be an object",
  "isValid": false,
}
`;

exports[`conf.registries is an empty object 1`] = `
Object {
  "error": "Configuration is not valid: registries must contain at least one endpoint",
  "isValid": false,
}
`;

exports[`validator api 1`] = `
Object {
  "validateConfiguration": [Function],
}
`;
